### ISSUE TITLE
PHP 8.0 Mac Apple Silicon - configure: error: Please reinstall the gpgme distribution

### ISSUE THREAD

**Issue by tntsoft** [php-gnupg/php-gnupg#42]
`checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking for pkg-config... /opt/homebrew/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether cc accepts -g... yes
checking for cc option to enable C11 features... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... aarch64-apple-darwin21.2.0
checking host system type... aarch64-apple-darwin21.2.0
checking target system type... aarch64-apple-darwin21.2.0
checking for PHP prefix... /opt/homebrew/Cellar/php@8.0/8.0.14
checking for PHP includes... -I/opt/homebrew/Cellar/php@8.0/8.0.14/include/php -I/opt/homebrew/Cellar/php@8.0/8.0.14/include/php/main -I/opt/homebrew/Cellar/php@8.0/8.0.14/include/php/TSRM -I/opt/homebrew/Cellar/php@8.0/8.0.14/include/php/Zend -I/opt/homebrew/Cellar/php@8.0/8.0.14/include/php/ext -I/opt/homebrew/Cellar/php@8.0/8.0.14/include/php/ext/date/lib
checking for PHP extension directory... /opt/homebrew/Cellar/php@8.0/8.0.14/pecl/20200930
checking for PHP installed headers prefix... /opt/homebrew/Cellar/php@8.0/8.0.14/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for gawk... no
checking for nawk... no
checking for awk... awk
checking if awk is broken... no
checking for gnupg support... yes, shared
checking for gnupg files in default path... not found
configure: error: Please reinstall the gpgme distribution
`

**Running PHP 8.0**

I've install gnupg and gpgme with homebrew. Tried to pecl install gnupg a ton of times to no avail.
I've also tried to download the source code itself and compile, still the same above end result.

Any help would be greatly appreciated! Thank you!

**Comment by bukka** [comment]
It complains because it cannot find `gpgme.h`. It's basically this logic
```sh
  SEARCH_PATH="/usr/local /usr /opt"
  SEARCH_FOR="include/gpgme.h"
  SEARCH_FOR_ALT="include/gpgme/gpgme.h"
  if test -r $PHP_GNUPG/$SEARCH_FOR; then
    GNUPG_DIR=$PHP_GNUPG
  else
    AC_MSG_CHECKING([for gnupg files in default path])
    for i in $SEARCH_PATH ; do
      if test -r $i/$SEARCH_FOR; then
        GNUPG_DIR=$i
        AC_MSG_RESULT(found in $i)
      fi
      if test -r $i/$SEARCH_FOR_ALT; then
        GNUPG_DIR=$i
        AC_MSG_RESULT(found in $i)
      fi
    done
  fi
```
It means if `gpgme.h` is not in `/usr/local`, `/usr` or `/opt`, then you can specify in compilation using
```
./configure --with-gnupg=$GPGME_PATH
```
where `$GPGME_PATH` is the dir where the header is installed.

**Comment by tntsoft** [comment]
So, I got it working yesterday.

I ended up modifying config.m4 and adding the path "/opt/homebrew/include", but that wasn't enough.

I also had to add CFLAGS on configuration that pointed to the above path (as it couldn't find gpg-error.h) without the flags.

I feel this path and the new homebrew path should be added to the repo to prevent future problems for Apple users.

**Comment by bukka** [comment]
@tntsoft Could you pls verify if https://github.com/php-gnupg/php-gnupg/pull/43 fixes your issue?

**Comment by tntsoft** [comment]
@bukka It should.

**Comment by bukka** [comment]
You should not need to update CFLAGS if `/opt/homebrew` is in the search path. I just merged it to master so should hopefully work with master as well.

**Comment by RobertBoes** [comment]
@bukka Will this ever be released? I see the latest tag is 1.5.1, which doesn't include this fix, so running `pecl install gnupg` doesn't work, the only way would be to use it from source, which in turn could potentially have lots of differences from 1.5.1
